server/src/main/java/org/opensearch/search/query/QueryPhase.java

```diff
@@ -88,7 +89,9 @@ public class QueryPhase {
     private static final Logger LOGGER = LogManager.getLogger(QueryPhase.class);
     // TODO: remove this property
     public static final boolean SYS_PROP_REWRITE_SORT = Booleans.parseBoolean(System.getProperty("opensearch.search.rewrite_sort", "true"));
-    public static final QueryPhaseSearcher DEFAULT_QUERY_PHASE_SEARCHER = new DefaultQueryPhaseSearcher();
+    public static final QueryPhaseSearcher DEFAULT_QUERY_PHASE_SEARCHER = FeatureFlags.isEnabled(FeatureFlags.CONCURRENT_SEGMENT_SEARCH)
```

Comment:
The `DEFAULT_QUERY_PHASE_SEARCHER` should stay unchanged and should not be impacted by the  `FeatureFlags`

Reply:
My thought is eventually `ConcurrentQueryPhaseSearcher` will become default in core and there won't be a need for separate `DefaultQueryPhaseSearcher` hence keeping this assignment behind feature flag

Reply:
We are honoring the `QueryPhaseSearcher` if provided via a plugin. The `DEFAULT_QUERY_PHASE_SEARCHER` instance is used only when no `QueryPhaseSearcher` is explicitly provided.

Reply:
What I am trying to say here is that the feature flag got leaked everywhere and makes it difficult to understand what is going to happen. The decisions should be taken ideally in one place, in the `SearchModule`: check plugins -> check feature flag -> use one or none

Reply:
moved to `SearchModule`